--- a/drivers/bcma/driver_gpio.c
+++ b/drivers/bcma/driver_gpio.c
@@ -184,7 +184,11 @@ int bcma_gpio_init(struct bcma_drv_cc *c
 	chip->direction_input	= bcma_gpio_direction_input;
 	chip->direction_output	= bcma_gpio_direction_output;
 	chip->parent		= bus->dev;
+#if LINUX_VERSION_IS_GEQ(5,17,0)
 	chip->fwnode		= dev_fwnode(&cc->core->dev);
+#elif IS_BUILTIN(CONFIG_OF)
+	chip->of_node		= cc->core->dev.of_node;
+#endif
 
 	switch (bus->chipinfo.id) {
 	case BCMA_CHIP_ID_BCM4707:
